src/controllers/CommentsController.ts

```diff
@@ -0,0 +1,50 @@
+import { Request, Response } from 'express'
+import validateComment from '../validation/addComment'
+import { Comments, User } from '../db'
+import { IUserRequest } from 'interfaces/IUserRequest'
+import validateParams from '../validation/paramsId'
+
+export default class CommentsController {
+  // for getting all data
+  public static async index (req: Request, res: Response) {
+    const { eventId } = req.params
+    await validateParams({ id: eventId })
+    const allComments = await Comments.findAll({
+      where: { EventId: eventId },
+      include: {
+        model: User,
+        attributes: ['id', 'username', 'profileImg']
+      }
+    })
+    res.json({ message: 'Comments received successfully', data: allComments })
+  }
+
+  // for getting just on element of data
+  //  (like getting just one event may be in event details)
+  public static async show (req: Request, res: Response) {
+  // code here
+  }
+
+  // for storing new data
+  public static async store (req: IUserRequest, res: Response) {
+    const { eventId } = req.params
+    await validateParams({ id: eventId })
+    const { image, content } = req.body
+
+    await validateComment({ image, content })
```

Comment:
When you have 2 or more independent async function (ex. when you don't need the result of one as an input for the other), its better to put them together in `Promise.All` so they can launch together then take less time to go the next line

```suggestion
    const { image, content } = req.body
    await Promise.all([validateParams({ id: eventId }), validateComment({ image, content })]) 
```

Reply:
Wow, thanks for these valuable notes ♥